components/compliance-service/reporting/relaxting/migrations.go

```diff
@@ -41,6 +42,31 @@ func StoreExists(client *elastic.Client, indexName string) (bool, error) {
 	return exists, nil
 }
 
+func cleanupEmptyIndicesForPrefix(client *elastic.Client, prefix string, migratable esMigratable) ([]string, error) {
+	catIndicesResponse, err := client.CatIndices().Index(prefix).Do(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	indices := make([]string, len(catIndicesResponse))
+
+	for _, catIndicesResponseRow := range catIndicesResponse {
+		indexName := catIndicesResponseRow.Index
+		logrus.Debugf("\nIndex: %s ", indexName)
+
+		dateAsStr := strings.SplitAfterN(indexName, "-", 4)
+		dateToRemove, err := time.Parse("2006.01.02", dateAsStr[len(dateAsStr)-1])
```

Comment:
Is there a chance that the `dateAsStr` will have zero length?

Reply:
^^ that's a good point. probably worth a guard on that

Reply:
good catch.  If that dateAsStr is zero len, we now log a warning and move on.